source/DataElement/AMDSCommandManager.cpp

```diff
@@ -0,0 +1,101 @@
+#include "AMDSCommandManager.h"
+
+AMDSCommandManager::~AMDSCommandManager()
+{
+	commands_.clear();
+	commandHash_.clear();
+	commandHexMapping_.clear();
+}
+
+AMDSCommand AMDSCommandManager::amdsCommand(int commandID) const {
+	return commandHash_.value(commandID);
+}
+
+AMDSCommand AMDSCommandManager::amdsCommand(const QString &commandString) const{
+	return commandHexMapping_.valueF(commandString);
+}
+
+AMDSCommand AMDSCommandManager::amdsCommandFromHex(const QString &hex) const
+{
+	QString command = commandFromHex(hex);
+	return amdsCommand(command);
+}
+
+int AMDSCommandManager::commandId(const QString &command) const
+{
+	return amdsCommand(command).commandId();
+}
+
+QString AMDSCommandManager::commandFromHex(const QString &hexString) const{
+	AMDSCommand other(-1, hexString, "", -1);
+	return commandHexMapping_.valueR(other);
+}
+
+/// ===================== implementation of AMDSCommand ==========================
+
+AMDSCommand AMDSCommand::decodeAndInstantiateAMDSCommand(QDataStream *dataStream)
+{
+	AMDSCommand commandDef;
+	commandDef.readFromDataStream(dataStream);
+	return commandDef;
+}
+
+bool AMDSCommand::encodeAndwriteAMDSCommand(QDataStream *dataStream, AMDSCommand command)
+{
+	return command.writeToDataStream(dataStream);
+}
+
+AMDSCommand::AMDSCommand(int commandId, const QString &hex, const QString &command, int timeout, int retries, QStringList responseHexes)
+{
+	commandId_= commandId;
+	hex_ = hex;
+	command_ = command;
+	timeout_ = timeout;
+	retries_ = retries;
+	responseHexes_ = responseHexes;
+}
+
+bool AMDSCommand::writeToDataStream(QDataStream *dataStream)
+{
+	// we only need to read/write commandId and command string for the AMDSCommand
+	*dataStream << commandId();
+//	*dataStream << hex();
```

Comment:
So far, I didn't see the necessary to send those getters. But in case we need them later, I simply commented off the codes. We can remove them permanently later when we are wrapping up the AMDS project